guess-the-country/country-backend/country/api.py

```diff
@@ -62,7 +62,7 @@
                     (35.2198646, 31.7956043), (35.2152225, 31.7936072), (35.2118609, 31.7921391), (35.2089856, 31.792431), (35.2044278, 31.7923225), (35.2030115, 31.7910093), (35.2009731, 31.7907722), (35.1996133, 31.7893066), (35.1999968, 31.7878228), (35.1954584, 31.7833671), (35.1901155, 31.7826922), (35.1869949, 31.7788226), (35.1844951, 31.7763532), (35.1817982, 31.7716133)]
     }
 
-dict_country = [tel_aviv, berlin, hamburg, jerusalem]
+dict_country = [tel_aviv, jerusalem, berlin, hamburg]
```

Comment:
```suggestion
all_cities = [tel_aviv, jerusalem, berlin, hamburg]
cities = {city["city"]: city for city in all_cities}
```

1. Ist das bisher kein `dict`, sondern eine `list` ;-)
2. Ist es unnötig, den Variablentyp mit in den Namen der Variable zu nehmen
3. Vermeidet ein Aufrufen per Name der Stadt (muss dann natürlich unten angepasst werden) ein für allemal das Reihenfolgen-Problem :-)

Reply:
Wenn du ein Mapping von Modell-Output -> Stadt brauchst, würde ich das separat machen.

Also z.B. eine extra Funktion:
```python
MODEL_OUTPUT_MAP = ["tel_aviv", "jerusalem", "berlin", "hamburg"]
def decode_model_output(output: np.ndarray) -> str:
    return MODEL_OUTPUT_MAP[int(np.argmax(output))]
```

Das Heraussuchen der Koordinaten und das entschlüsseln des Modell-Outputs sind ja zwei logisch getrennte Probleme.

Reply:
Du hast natürlich mit allem recht :)
Zu der Output Funktion nur folgendes, ich hatte ja so eine MODEL_OUTPUT_MAP in einer älteren Version auch drinnen, dachte aber es wäre cool wenn man auf die selbe Array wie für die Koordinaten zugreift. Ist natürlich logisch getrennt aber bei dem Hinzufügen einer Stadt müsste man nicht soviele Stellen anfassen war mein Gedanke.

Reply:
Das stimmt, aber der Modell-Output gehört untrennbar zum Modell. D.h. wenn du das Modell änderst müsstest du das Koordinaten-Verzeichnis ändern, obwohl das überhaupt nichts mit dem Modell zu tun hat.

Daher in `model.predict` den Modell-Output in die "Sprache" deines Backends übersetzen und dann in `api` (idealerweise später mal ausgelagert in ein eigenes Modul) diese "Sprache" für die Streetview-Abfrage in Koordinaten übersetzen.

Reply:
Erledigt 👍